### DISCUSSION TITLE
👋 Welcome to Selldone's Storefront Application Discussion!

### DISCUSSION THREAD

**Discussion by pajuhaan** [selldone/storefront#1] [Announcements]
**Hello there, Selldone enthusiasts! 🌟**

Firstly, a bit about Selldone: We believe in replacing ∞ plugins & apps with just ONE! Our vision is to empower you to build, manage, and scale your business like the pros. With our enterprise-grade, no-code business operating system, you can handle millions of orders and automate your workflow seamlessly. 🚀

Our Manifest: We cannot imagine teams without coaches, revolutions without leaders, trees without roots, and businesses without an operating system! That's why we're reimagining online business for fast-growing companies and young entrepreneurs. 🌱

### Why are we here?
We’re using Discussions as a vibrant space to connect with other members of our community. Here's what we hope for:

- 🤔 **Ask Questions**: If there's something you're wondering about, ask away!
- 💡 **Share Ideas**: Got a brilliant idea or a suggestion? We're all ears!
- 🤝 **Engage**: Connect, discuss, and collaborate with other community members.
- 🤗 **Be Welcoming**: Let's create a positive and inclusive environment. Remember, we're building this community together 💪.

**Getting Started**: Dive right in by commenting below. Introduce yourself and share what you're doing or planning to do with Selldone's storefront application.

 [selldone.com](https://selldone.com/)

**Comment by nanohits** [comment]
For the local server does it have to be a linux OS?

**Reply by pajuhaan** [reply]
Sky's the limit! You have the freedom to **self-host** your webapp **on any server** you choose. It's as straightforward as copying and pasting files.  I'll be writing and sharing an article about this process asap.

**Reply by nanohits** [reply]
So how will this be tied for those who are on the Enterprise plans? Would you not be hosting this or are we supposed to be hosting these? Our main objective will be to use this as a 100% white label solution with no references to selldone.

**Reply by pajuhaan** [reply]
You have both options to **self-host** the storefront or **host your developed storefront on Selldone** (no need for any server). By redesigning the storefront from scratch or starting from this ready-to-build project, enterprise licenses have the option to make it **fully whitelabeled**. So, Selldone acts like Stripe -in payment- to handle **APIs related to commerce functionality**. So, you can add more independent functionality to your store without any limits. 

Please read this and ask questions; I will add more information to clarify issues there: [Deploy & Hosting](https://github.com/selldone/storefront/wiki/Deploy-&-Hosting).

**Comment by josodigital** [comment]
Hi Pajuhaan,
I was asked by your colleague Alizera to post of this board for an ansswer to my question. I have posted under "General" but no answers. So posting here. I hope I can get a reply this time. Many thanks.

Please see my query below:

Hi there,
I am an Enterprise Selldone user and a Partner. My understanding is that the new SDK was meant to make life easier in terms of storefront customizations. When I read your instructions on Github, it was clear I needed to install this on a server. Thankfully, I happen to have my own server. So I did install successfully using npm install, npm setup.

I run into trouble on the "Step 3: Begin Development" . when I run npm serve. It says under Network: Unavailable? When I try to open https://myserveraddress:8080/ I get a an error and a blank page. However when I look at the code on that page, I see selldone data that is not displayed by the browser?

So I have been looking for the location to change the Network address to my server IP address?

This is where I am at and I need help. I understand you might be working on solution?

I think given I have engaged in this, the sooner you can help me the better. I am happy to jump on a zoom call to make this happen?

Looking forward to your reply.

Kind regards,

Fleman

**Reply by nanohits** [reply]
It would be a good idea if they made some videos to make things real easy as not everyone are very technical but reasonably technical. I intend to install this next week when I am back from vacation.

**Reply by pajuhaan** [reply]
Hi @josodigital 
This project is developed using the **Vue framework**, which is a JavaScript framework for **client-side** development. 

This means the final product will be a web app, includes JavaScript, CSS, and other assets. To get started with this project, you'll need to **clone and build it on your PC**. 

Once built, you can **publish it using Selldone CLI**, available at https://www.npmjs.com/package/selldone-cli. This tool allows you to deploy the project either on Selldone or on your own server (_with server support coming soon through Selldone CLI_). 

Additionally, the command 'yarn serv' is used to set up a development environment on your PC, enabling real-time development and testing of the project.

**Reply by pajuhaan** [reply]
@nanohits Hi, absolutely, we will publish videos as soon as possible after releasing the stable project, along with the libraries and CLI. Currently, we are focusing on developing a suite of tools to simplify and streamline the process of creating, developing, and publishing.

**Reply by nanohits** [reply]
It will work for Mac too right as I dont have a PC.
